client/my-sites/domains/domain-management/name-servers/with-domain-nameservers.js

```diff
@@ -14,62 +14,31 @@ import useDomainNameserversQuery from 'calypso/data/domains/nameservers/use-doma
 import useUpdateNameserversMutation from 'calypso/data/domains/nameservers/use-update-nameservers-mutation';
 import { errorNotice, successNotice } from 'calypso/state/notices/actions';
 
-const useSuccessNotice = ( isSuccess ) => {
-	const showNotice = useRef();
-	const dispatch = useDispatch();
-	const translate = useTranslate();
-
-	useEffect( () => {
-		showNotice.current = () => {
-			dispatch(
-				successNotice( translate( 'Yay, the name servers have been successfully updated!' ), {
-					duration: 5000,
-					id: `nameserver-update-notification`,
-				} )
-			);
-		};
-	}, [ dispatch, translate ] );
-
-	useEffect( () => {
-		isSuccess && showNotice.current();
-	}, [ isSuccess ] );
-};
-
-const useErrorNotice = ( isError, error ) => {
-	const showNotice = useRef();
-	const dispatch = useDispatch();
-	const translate = useTranslate();
-
-	useEffect( () => {
-		showNotice.current = () => {
-			const defaultMessage = translate( 'An error occurred while updating the nameservers.' );
-			dispatch(
-				errorNotice( error?.message ?? defaultMessage, {
-					duration: 5000,
-					id: `nameserver-update-notification`,
-				} )
-			);
-		};
-	}, [ dispatch, translate, error ] );
-
-	useEffect( () => {
-		isError && showNotice.current();
-	}, [ isError ] );
-};
-
 const withDomainNameservers = createHigherOrderComponent( ( Wrapped ) => {
 	const WithDomainNameservers = ( props ) => {
 		const { selectedDomainName } = props;
+		const dispatch = useDispatch();
+		const translate = useTranslate();
 		const { data, isLoading, isError, error } = useDomainNameserversQuery( selectedDomainName );
-		const {
-			updateNameservers,
-			isSuccess: isUpdateSuccess,
-			isError: isUpdateError,
-			error: updateError,
-		} = useUpdateNameserversMutation( selectedDomainName );
-
-		useSuccessNotice( isUpdateSuccess );
-		useErrorNotice( isUpdateError, updateError );
+		const { updateNameservers } = useUpdateNameserversMutation( selectedDomainName, {
+			onSuccess() {
+				dispatch(
+					successNotice( translate( 'Yay, the name servers have been successfully updated!' ), {
+						duration: 5000,
+						id: `nameserver-update-notification`,
+					} )
+				);
+			},
+			onError( err ) {
+				const defaultMessage = translate( 'An error occurred while updating the nameservers.' );
+				dispatch(
+					errorNotice( err?.message ?? defaultMessage, {
```

Comment:
I'm not convinced we need to do `err?` here - is there any way it will be nullish?